src/api/subscriptions/setTopicMute.js

```diff
@@ -5,12 +5,12 @@ import { apiPatch } from '../apiFetch';
 /** See https://zulip.com/api/mute-topic */
 export default async (
   auth: Auth,
-  stream: string,
+  streamId: number,
   topic: string,
   value: boolean,
 ): Promise<ApiResponse> =>
   apiPatch(auth, 'users/me/subscriptions/muted_topics', {
-    stream,
+    stream_id: streamId,
```

Comment:
This will change what interface we're assuming from the server when we make a request there. When we do that, we want to be sure to say in the commit message why we believe the new interface will work.

Following the link to #3244, it looks like this new way is accepted starting with server 2.0. We're planning to desupport servers older than that soon, but we're not quite ready to do so -- #4750 added a banner to warn people on such old servers that it's time to upgrade, and we'll want to have the release with that banner out for a bit before having a further release that actually stops supporting the older servers.

So let's hold this commit off for now, leaving it out of this PR.